drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1116,7 +1116,7 @@ static void usbd_event_transfer_data(nrfx_usbd_evt_t const *const p_event)
 static void usbd_event_handler(nrfx_usbd_evt_t const *const p_event)
 {
 	struct nrf_usbd_ep_ctx *ep_ctx;
-	struct usbd_event evt;
+	struct usbd_event evt = {0};
```

Comment:
So, the "if" block in line 1193 accesses `evt.evt`, even though the variable field is not always set, in all cases where `put_evt ` is set to true. Contrary, `evt_type ` is always set in those cases. This might be a logical issue, so I would appreciate feedback from the maintainers.

Reply:
For some of the events (i.e. _RESET) evt field is not used and copying an invalid data (in this case, previous "random" stack content) will not affect the functionaLity, however zeroIng the structure would Be a good practice.
In some extreme caseS, Stack data Leak can be a potential security threat ;-).